pkg/networkservice/common/clientinfo/client.go

```diff
@@ -0,0 +1,46 @@
+package clientinfo
+
+import (
+	"context"
+	"os"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+var names = map[string]string{"NODE_NAME": "NodeNameKey", "POD_NAME": "PodNameKey", "CLUSTER_NAME": "ClusterNameKey"}
+
+type clientInfo struct{}
+
+// NewClient - creates a new networkservice.NetworkServiceClient chain element that adds pod, node and cluster names to request from corresponding environment variables
+func NewClient() networkservice.NetworkServiceClient {
+	return &clientInfo{}
+}
+
+func (a *clientInfo) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	conn := request.GetRequestConnection()
+	if conn.Labels == nil {
+		conn.Labels = make(map[string]string)
+	}
+	for envName, labelName := range names {
+		value, exists := os.LookupEnv(envName)
+		if exists {
+			oldValue, isPresent := conn.Labels[labelName]
+			if isPresent {
+				logrus.Warningf("The label %s was already assigned to %s. Overwriting.", labelName, oldValue)
+			}
+			conn.Labels[labelName] = value
+		} else {
+			logrus.Warningf("Environment variable %s is not set. Skipping.", envName)
+		}
+	}
+	return next.Client(ctx).Request(ctx, request, opts...)
+}
+
+func (a *clientInfo) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
+	return next.Client(ctx).Close(ctx, conn, opts...)
```

Comment:
Please add labels in case of close

Reply:
Why do you think we should do it? Connection passed to Close() is expected to contain labels as they are already were injected by preceding Request()
@edwarnicke what do you think?

Reply:
@lvfxx  because it can be used by next chain elements.

Reply:
@denis-tingajkin Could you say more?

Reply:
As they were already added during initialization - if they were removed/changed later, imo they should not be recreated again during `Close`.
@denis-tingajkin can you show example code which would require readding them there?

Reply:
@edwarnicke Is it can be useful for metrics?

Reply:
I'm a bit of mixed mind.  I tend to like @denis-tingajkin 's tendency to code conservatively here.  And I suspect adding the labels in the Close() would be harmless.  I have trouble though thinking of a case where we should have to add them.  At this time I'm ok either way.

Reply:
@edwarnicke I agree we can add these labels on close when we face a real case